text/3750-cfg-os-version-min.md

```diff
@@ -0,0 +1,130 @@
+* Certain features cannot be dynamically detected and thus limit possible implementations. The libc crate must use [a raw syscalls on Android for `accept4`](https://github.com/rust-lang/libc/pull/1968), because this was only exposed in libc in version 21 of the Android API.  Additionally libstd must dynamically load `signal` for all versions of Android despite it being required only for versions 19 and below. In the future there might be similar changes where there is no way to implement a solution for older versions.
+* Trying to compile code with an implicit dependency on a API version greater than what is supported by the target platform leads to linker errors. For example, the `x86_64-pc-windows-msvc` target's rustc implementation requires `SetThreadErrorMode` which was introduced in Windows 7. This means trying to build the compiler on older versions of Windows will fail with [a less than helpful linker error](https://github.com/rust-lang/rust/issues/35471).
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Rust targets are often thought of as monoliths.
+The thought is that if you compile a binary for that target, that binary should be able to run on any system that fits that target's description.
+However, this is not actually true.
+For example, when compiling for `x86_64-pc-windows-msvc` and linking with the standard library, my binary has implicitly taken a dependency on a set of APIs that Windows exposes for certain functionality.
+If I try to run my binary on older systems that do not have those APIs, then my binary will fail to run.
+When compiling for a certain target, you are therefore declaring a dependency on a minimum target API version that you rely on for your binary to run.
+
+Each standard library target uses a sensible minimum API version. for `x86_64-pc-windows-msvc` the minimum API version is "10.0.10240" which corresponds to Windows 10's initial release.
+For `x86_64-win7-pc-windows-msvc` the minimum API version is "6.1.7600" which corresponds to Windows 7.
+However, inferring the API version from the target name isn't ideal especially as it can change over time.
+
+Instead you use the `os_version_min` predicates to specify the minimum API levels of various parts of the operating system.  For example:
+
+* `os_version_min(“windows”, <string>)` would test the [minimum build version](https://gaijin.at/en/infos/windows-version-numbers) of Windows.
+* `os_version_min(“libc”, <string>)` would test the version of libc.
+* `os_version_min(“kernel”, <string>)` would test the version of the kernel.
```

Comment:
Throughout this PR you use "smart quotes" `“..”` instead of `".."`. This means that users wouldn't be able to copy+paste the code.

Reply:
Yeah, I'm not sure if that was present in the original or if my editor inserted it on copy/paste but either way it's not good.

I'm doing a fairly substantial rewrite of this RFC to address other comments so this will be fixed in the update version.